File: VB/EditableGroupRow/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms

Namespace EditableGroupRow

    Public Partial Class Form1
        Inherits Form

        Private dt As DataTable = New DataTable()

        Public Sub New()
            InitializeComponent()
            dt.Columns.Add("Name", GetType(String))
            dt.Columns.Add("Age", GetType(Integer))
            dt.Rows.Add("Name 1", 25)
            dt.Rows.Add("Name 2", 25)
            dt.Rows.Add("Name 3", 35)
            dt.Rows.Add("Name 4", 55)
            dt.Rows.Add("Name 5", 55)
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            gridControl1.DataSource = dt
            Dim re As GroupRowEdit = New GroupRowEdit(gridView1)
        End Sub
    End Class
End Namespace

File: CS/EditableGroupRow/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using DevExpress.Utils.Menu;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.XtraEditors;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Columns;


namespace EditableGroupRow
{
    
    public partial class Form1 : Form
    {
        DataTable dt = new DataTable();

        public Form1()
        {
            InitializeComponent();
            dt.Columns.Add("Name", typeof(string));
            dt.Columns.Add("Age", typeof(int));
            dt.Rows.Add("Name 1", 25);
            dt.Rows.Add("Name 2", 25);
            dt.Rows.Add("Name 3", 35);
            dt.Rows.Add("Name 4", 55);
            dt.Rows.Add("Name 5", 55);
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            gridControl1.DataSource = dt;
            GroupRowEdit re = new GroupRowEdit(gridView1);
        }

    }
}